lib/plugins/aws/package/lib/mergeIamTemplates.js

```diff
@@ -87,30 +87,39 @@ module.exports = {
       .Statement;
 
     let hasOneOrMoreCanonicallyNamedFunctions = false;
+    const disableLogsCanonicallyNamedFunctions = [];
 
     // Ensure policies for functions with custom name resolution
     this.serverless.service.getAllFunctions().forEach(functionName => {
-      const { name: resolvedFunctionName } = this.serverless.service.getFunction(functionName);
+      const {
+        name: resolvedFunctionName,
+        disableLogs: isDisableLogsOpt,
+      } = this.serverless.service.getFunction(functionName);
       if (!resolvedFunctionName || resolvedFunctionName.startsWith(canonicalFunctionNamePrefix)) {
+        if (isDisableLogsOpt) {
+          disableLogsCanonicallyNamedFunctions.push(resolvedFunctionName);
+        }
         hasOneOrMoreCanonicallyNamedFunctions = true;
```

Comment:
Let's put it in `else` clause (e.g. if all functions named canonically we should just stay at not creating "Allow" policies